[Hoon Chee Ping] Duke Increments

hooncp opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

hooncp committed:
Merge branch 'gradle'

hooncp committed:
level 1 of Duke program

hooncp committed:
Revert "level 1 of Duke program"

This reverts commit f451921820d51d4f6ac4de74e4e5c2c0af175d72.

hooncp committed:
level 1 of Duke

hooncp committed:
Completed level 2 of Duke

hooncp committed:
added Task class and integrated with Duke class

hooncp committed:
changed equals to equalsIgnoreCase

hooncp committed:
added Todo, Deadline and Event classes

hooncp committed:
integrated Event Todo and Deadline to Duke and complete level 5

hooncp committed:
added in InvalidInputException and IndexOutofBoundsException

hooncp committed:
added in more exceptions

hooncp committed:
added in delete function and respective exceptions

hooncp committed:
added text-ui-test

hooncp committed:
added Storage to take care of loading tasks and saving tasks

hooncp committed:
added in date and time for level 8

hooncp committed:
Merge branch 'branch-Level-7'

hooncp committed:
Merge branch 'branch-Level-8'

hooncp committed:
updated .gitignore to ignore testfiles

hooncp committed:
added in UI class for better oop

hooncp committed:
improved UI and fixed minor null bugs

hooncp committed:
improved OOP: addded in TaskList

hooncp committed:
improved OOP: added read command to UI

hooncp committed:
Fixed datetime storage into txt file bug

hooncp committed:
improved OOP: added in Command class and Parser

hooncp committed:
added OOP and improved Command and Parser

hooncp committed:
Shifted more print messages to ui

hooncp committed:
Refactor UI

hooncp committed:
add j-unit testing

hooncp committed:
added in javadocs

hooncp committed:
Fixed minor styling errors

hooncp committed:
add new find feature for level 9

hooncp committed:
Merge tag 'A-JavaDoc'

hooncp committed:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

hooncp committed:
Merge Level 9 with master

hooncp committed:
Merge pull request #1 from hooncp/branch-A-JavaDoc

Branch a java doc

hooncp committed:
Fix javadocs

hooncp committed:
Add more javadocs

hooncp committed:
Package java files

ryanYtan commented:
## Code

### `duke.task`
Good inheritance done on the classes here, encapsulation also done by `TaskList`

### `duke.command`
Instead of an `abstract` method `boolean isExit()` that has to be implemented by all subclasses of `Command`, you may want to implement `boolean isExit()` as non-abstract in the `Command` base class that returns `false`, and then have `ByeCommand` override it to return `true` instead.

### `Duke.java`
* In `Duke.java`, you can combine the `catch` statements together using the `|` operator:  

   ```
   try {
       // ...
   } catch (Exception1 | Exception2 | Exception3 e) {
       // ...
   }
   ```

### `Storage.java`
Not sure if `rewriteData()` will write to the file if it doesn't exist, I remember having some trouble with that on my end.

### `Ui.java`
My approach to `Ui` was to have a general `public void print(String... args)` function so any method with access to `Ui` could print whatever they wanted. This makes it so that you don't have to change the `Ui` class if one of your `Command` subclasses needs to print a different thing (also because I had some additional formatting provided by the `Ui` class), or if you need to implement another `Command` subclass.

Not sure if it's any better than your method (since my method still requires a reference to a `Ui` instance *anyway*, and you'll be hardcoding the strings anyway). No comments otherwise.

### `Parser.java`
* Using `switch...case` may look cleaner rather than continuous `if else`
* It might not be recommended to import using `duke.command.*` (even though you actually have to import everything...)

## Style
Style guide followed, JavaDocs done, etc. I spotted one very tiny mistake where in `Storage.java`, there is no space between `IOException` and opening `{` on line 31, but that's about it.

hooncp committed:
Add javaFX

hooncp committed:
Add javaFX

(cherry picked from commit 68009e7d6c2cb8345f01b96918d0190ebd803ea2)

hooncp committed:
Add FXML to Duke and repackage some classes

(cherry picked from commit e8ea5f05aa46cb671bfbb799053c8d0b72070715)

hooncp committed:
Add FXML to Duke and repackage some classes

Yen-Peng commented:
### Code Review
**Duke.java**

- Good packaging into command/task.

**Command.java**

- Good abstraction of command class and inheritance for child classes.

**Todo.java**

- May be better to insert a space between `Task` and `{` in line 8.

**Code**

- Good documentation of classes and methods.

hooncp committed:
Add greet to javaFX and output from user input.

hooncp committed:
Add greet to javaFX and output from user input.

(cherry picked from commit 64604918e54de897efa0b010643d9c3d98055246)

hooncp committed:
Refactor code

hooncp commented:
> you may want to implement boolean isExit() as non-abstract in the Command base class that returns false, and then have ByeCommand override it to return true instead.

 Refactored code for command abstract class.

>combine the catch statements together using the | operator:

 Completed.

>Parser.java
It might not be recommended to import using duke.command.* (even though you actually have to import everything...)

 Updated.

hooncp committed:
Merge branch 'branch-Level-10'

hooncp committed:
Add Shadow into build

hooncp committed:
Add assertions to Duke

hooncp committed:
Merge pull request #4 from hooncp/branch-A-Assertions

Add assertions to Duke

hooncp committed:
Refactor code quality (#5)

hooncp committed:
Add additional functionalities to Delete and Done Command

hooncp committed:
Add CheckStyle to Duke

hooncp committed:
Refactor code

hooncp committed:
Re-package java files

hooncp committed:
Improve code quality

hooncp committed:
Improve UI

hooncp committed:
Update README.md

hooncp committed:
Fix minor errors

hooncp committed:
Merge branch 'master' of https://github.com/hooncp/duke

hooncp committed:
Add ui.png file

hooncp committed:
Delete ui.png

hooncp committed:
Reload Ui.png

hooncp committed:
Set theme jekyll-theme-slate

hooncp committed:
Update README.md

hooncp committed:
Fix cross platform issues and bugs

hooncp committed:
Fix minor bugs

hooncp committed:
Update README.md

hooncp committed:
Update minor changes to README.md
